pkg/controller/podgc/gc_controller_test.go

```diff
@@ -135,90 +150,203 @@ func TestGCTerminated(t *testing.T) {
 
 		gcc.gc()
 
-		pass := true
-		for _, pod := range deletedPodNames {
-			if !test.deletedPodNames.Has(pod) {
-				pass = false
-			}
-		}
-		if len(deletedPodNames) != len(test.deletedPodNames) {
-			pass = false
-		}
-		if !pass {
-			t.Errorf("[%v]pod's deleted expected and actual did not match.\n\texpected: %v\n\tactual: %v", i, test.deletedPodNames, deletedPodNames)
+		if pass := compareStringSetToList(test.deletedPodNames, deletedPodNames); !pass {
+			t.Errorf("[%v]pod's deleted expected and actual did not match.\n\texpected: %v\n\tactual: %v",
+				i, test.deletedPodNames.List(), deletedPodNames)
 		}
 	}
 }
 
 func TestGCOrphaned(t *testing.T) {
-	type nameToPhase struct {
-		name  string
-		phase v1.PodPhase
+	type podMeta struct {
```

Comment:
i don't see the value of having podMeta type here, and first creating podmeta and then creating Pod from it.
Instead, just:
- create a makePod function (sth like:
```
makePod := func(name, nodeName string, phase v1.PodPhase) *v1.Pod {
  return &v1.Pod{
    ObjectMeta: metav1.ObjectMeta{Name: name},
    Spec:       v1.PodSpec{NodeName: nodeName},
    Status:     v1.PodStatus{Phase: phase},
  }
}
```
and use it below (i.e. change pods to be of type []*v1.Pod

Reply:
OK

Reply:
Done